### ISSUE TITLE
Temporary variables can appear to be in scope when they aren't

### ISSUE THREAD

**Issue by joningold** [inkle/ink#421]
The following:

```
-> a(1) -> a.b 

== a(temporary) 
    ->-> 
= b 
    { temporary } 
    -> END 
```

generates an ugly run-time crash (can't find temporary variable in context -1), but the compiler doesn't pick it up as a problem. I'm not sure what I'd hope for from this - I can see that coping with this is pretty difficult!

**Comment by joningold** [comment]
Possibly the same, possibly only related, the following also results in a "can't find temp" crash.

```
-> a 

=== a 
    ~ temp x = 1 
    -> b ->

= b 
    { x } 
    ->-> 
```

**Comment by joethephish** [comment]
Yes, even this example will compile but crash at runtime, unfortunately:

```
{x}
~ temp x = 5
```

While it's easy to see that that is wrong, it's less easy to check for more complex cases like the ones you gave above.

As part of the wider move to help ink cope better with missing variables and story changes (https://github.com/inkle/ink/issues/195), this version will soon not-crash, and instead use a default value - zero. (But spit out a warning which can be caught and logged by the game.)

I'm not sure there's much else we can do about it?

**Comment by joningold** [comment]
Removing the "bug" label for now as, yeah, seems like there might not be much else we can do here.
